stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/OfflinePlugin.kt

```diff
@@ -1,288 +0,0 @@
-            .onGetMessageResult(result, cid, messageId, olderMessagesOffset, newerMessagesOffset)
-    }
-
-    override suspend fun onGetMessageError(
-        cid: String,
-        messageId: String,
-        olderMessagesOffset: Int,
-        newerMessagesOffset: Int,
-    ): Result<Message> = cid.cidToTypeAndId().let { (channelType, channelId) ->
-        logic.channel(channelType, channelId)
-            .onGetMessageError(cid, messageId, olderMessagesOffset, newerMessagesOffset)
-    }
-
-    /**
-     * Updates the messages locally and saves it at database.
-     *
-     * @param messages The list of messages to be updated in the SDK and to be saved in database.
-     * @param channelLogic [ChannelLogic].
-     */
-    private suspend fun updateAndSaveMessages(messages: List<Message>, channelLogic: ChannelLogic) {
-        channelLogic.upsertMessages(messages)
-        channelLogic.storeMessageLocally(messages)
-    }
-
-    /**
-     * Gets the channel logic for the channel of a message.
-     *
-     * @param message [Message].
-     */
-    private fun channelLogicForMessage(message: Message): ChannelLogic {
-        val (channelType, channelId) = message.cid.cidToTypeAndId()
-        return logic.channel(channelType, channelId)
-    }
-
-    /**
-     * Update the online state of a message.
-     *
-     * @param isOnline [Boolean].
-     */
-    private fun Message.updateMessageOnlineState(isOnline: Boolean): Message {
-        return this.copy(
-            syncStatus = if (isOnline) SyncStatus.IN_PROGRESS else SyncStatus.SYNC_NEEDED,
-            updatedLocallyAt = Date()
-        )
-    }
-
-    /**
-     * Updates a message that whose request (Edition/Delete/Reaction update...) has failed.
-     *
-     * @param chatError [ChatError].
-     */
-    private fun Message.updateFailedMessage(chatError: ChatError): Message {
-        return this.copy(
-            syncStatus = if (chatError.isPermanent()) {
-                SyncStatus.FAILED_PERMANENTLY
-            } else {
-                SyncStatus.SYNC_NEEDED
-            },
-            updatedLocallyAt = Date(),
-        )
-    }
-
-    override suspend fun onHideChannelPrecondition(
-        channelType: String,
-        channelId: String,
-        clearHistory: Boolean,
-    ): Result<Unit> =
-        logic.channel(channelType, channelId).onHideChannelPrecondition(channelType, channelId, clearHistory)
-
-    override suspend fun onHideChannelRequest(channelType: String, channelId: String, clearHistory: Boolean): Unit =
-        logic.channel(channelType, channelId).onHideChannelRequest(channelType, channelId, clearHistory)
-
-    override suspend fun onHideChannelResult(
-        result: Result<Unit>,
-        channelType: String,
-        channelId: String,
-        clearHistory: Boolean,
-    ): Unit = logic.channel(channelType, channelId).onHideChannelResult(result, channelType, channelId, clearHistory)
-
-    internal fun clear() {
```

Comment:
Plugins are stateless. If a dependency of a plugin contains state, then we should deal with this state in the dependency and not bring the state to the whole plugin.